infra/terraform/global/iam.tf

```diff
@@ -25,3 +25,76 @@ EOF
 resource "aws_iam_access_key" "auth0_ses" {
     user = "${aws_iam_user.auth0_ses.name}"
 }
+
+resource "aws_iam_role" "softnas" {
+  name = "SoftNAS_HA_IAM"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "softnas" {
+  name = "softnas_ha_policy"
+  role = "${aws_iam_role.softnas.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "Stmt1444200186000",
+      "Effect": "Allow",
+      "Action": [
+        "ec2:ModifyInstanceAttribute",
+        "ec2:DescribeInstances",
+        "ec2:CreateVolume",
+        "ec2:DeleteVolume",
+        "ec2:CreateSnapshot",
+        "ec2:DeleteSnapshot",
+        "ec2:CreateTags",
+        "ec2:DeleteTags",
+        "ec2:AttachVolume",
+        "ec2:DetachVolume",
+        "ec2:DescribeInstances",
+        "ec2:DescribeVolumes",
+        "ec2:DescribeSnapshots",
+
+        "aws-marketplace:MeterUsage",
+
+        "ec2:DescribeRouteTables",
+        "ec2:DescribeAddresses",
+        "ec2:DescribeTags",
+        "ec2:DescribeInstances",
+        "ec2:ModifyNetworkInterfaceAttribute",
+        "ec2:ReplaceRoute",
+        "ec2:CreateRoute",
+        "ec2:DeleteRoute",
+        "ec2:AssociateAddress",
+        "ec2:DisassociateAddress",
+
+        "s3:CreateBucket",
+        "s3:Delete*",
+        "s3:Get*",
+        "s3:List*",
+        "s3:Put*"
+      ],
+      "Resource": [
+        "*"
```

Comment:
I guess there is no way to be more specific in this case. Worth bearing in mind that with this policy it's possible to do a lot of stuff (didn't read the rest and not familiar with SoftNAS, so apologies if this is actually the minimum required).